xcp_d/cli/parser.py

```diff
@@ -845,6 +949,59 @@ def _validate_parameters(opts, build_log, parser):
     if opts.custom_confounds:
         opts.custom_confounds = str(opts.custom_confounds.resolve())
 
+    # Define parameters based on the mode
+    if opts.exact_time == "none":
+        opts.exact_time = []
+
+    if opts.mode == "abcdbids":
+        opts.despike = False if opts.despike == "auto" else opts.despike
+        opts.cifti = True if opts.cifti == "auto" else opts.cifti
+        opts.process_surfaces = True if opts.process_surfaces == "auto" else opts.process_surfaces
+        opts.dcan_qc = True if opts.dcan_qc == "auto" else opts.dcan_qc
+        opts.exact_time = [300, 480] if opts.exact_time == "auto" else opts.exact_time
+        opts.combineruns = True if opts.combineruns == "auto" else opts.combineruns
+    elif opts.mode == "hbcd":
+        opts.despike = False if opts.despike == "auto" else opts.despike
+        opts.cifti = True if opts.cifti == "auto" else opts.cifti
+        opts.process_surfaces = True if opts.process_surfaces == "auto" else opts.process_surfaces
+        opts.dcan_qc = True if opts.dcan_qc == "auto" else opts.dcan_qc
+        opts.exact_time = [300, 480] if opts.exact_time == "auto" else opts.exact_time
+        opts.combineruns = True if opts.combineruns == "auto" else opts.combineruns
```

Comment:
@tjhendrickson can you weigh in on this and note any other parameters that should be set here?

Reply:
Hi @tsalo, thanks for sending along. I am not super familiar with the code base, can you break down these details into human readable form?

Reply:
Absolutely. Here's the breakdown:

1. despiking is disabled
2. CIFTI processing is enabled
3. Surfaces in fsnative space will be warped to fsLR and parcellated morphometrics (e.g., sulcal depth) will be outputtted
4. DCAN QC-related outputs (e.g., interpolated, denoised BOLD data; the BrainSprites for the executive summary; the dcan hd5 motion file) will be created
5. Correlation matrices from 5 minutes and 8 minutes will be generated
6. Results (BOLD data and parcellated time series) will be concatenated across runs

Reply:
Set input type to "nibabies".
Set head radius to "auto".
Set despike to True.